tactics/evar_tactics.ml

```diff
@@ -26,17 +25,12 @@ module NamedDecl = Context.Named.Declaration
 (* Instantiation of existential variables *)
 (******************************************)
 
-let depends_on_evar sigma evk _ (pbty,_,t1,t2) =
-  try Evar.equal (head_evar sigma t1) evk
-  with NoHeadEvar ->
-  try Evar.equal (head_evar sigma t2) evk
-  with NoHeadEvar -> false
-
 let define_and_solve_constraints evk c env evd =
   if Termops.occur_evar evd evk c then
     Pretype_errors.error_occur_check env evd evk c;
+  let () = ignore (extract_all_conv_pbs evd) (* ensure that there is no pending modifications - ideally the case *) in
```

Comment:
AFAIU this function cannot raise an exception, what are you intending to do here?

Reply:
I hallucinated. I meant `let (evd, _) = extract_all_conv_pbs evd` so as to reset the `last_mods` field. Sorry.